### PR TITLE
.Net Agents - Add Integration Test

### PR DESCRIPTION
**Pull request by crickman**
### Motivation and Context

Adding integration test to validate interaction of `ChatCompletionAgent` and `OpenAIAssistantAgent` when using plug-ins.

### Description

Validate functionality in alignment with a recently found Python bug on how tool-call results are synchronized across agent types.

Also unify namespaces

### Contribution Checklist

- [X] The code builds clean without any errors or warnings
- [X] The PR follows the [SK Contribution Guidelines](https://github.com/microsoft/semantic-kernel/blob/main/CONTRIBUTING.md) and the [pre-submission formatting script](https://github.com/microsoft/semantic-kernel/blob/main/CONTRIBUTING.md#development-scripts) raises no violations
- [X] All unit tests pass, and I have added new tests where possible
- [X] I didn't break anyone :smile:

### CODE REVIEW

File: dotnet/src/IntegrationTests/Agents/MixedAgentTests.cs [hunk 1]
```diff
@@ -0,0 +1,161 @@
...
+            this.CreateChatCompletionKernel(openAISettings),
+            new(openAISettings.ApiKey),
+            openAISettings.ModelId);
+    }
+
+    /// <summary>
+    /// Integration test for <see cref="OpenAIAssistantAgent"/> using function calling
+    /// and targeting Azure OpenAI services.
+    /// </summary>
+    [Fact/*(Skip = "No supported endpoint configured.")*/]
```
Comment (dmytrostruk) [inline dotnet/src/IntegrationTests/Agents/MixedAgentTests.cs:48]: Should we remove comment here?
```suggestion
    [Fact]
```

**Review by dmytrostruk — APPROVED** [review]

**Review by TaoChenOSU — APPROVED** [review]

**Review by dmytrostruk — APPROVED** [review]

File: dotnet/src/IntegrationTests/Agents/MixedAgentTests.cs [hunk 2]
```diff
@@ -0,0 +1,162 @@
...
+            await this.InvokeAgentAsync(chat, chatAgent, "What is the special soup?", "Clam Chowder");
+            await this.InvokeAgentAsync(chat, assistantAgent, "What is the special drink?", "Chai Tea");
+        }
+        finally
+        {
+            await assistantAgent.DeleteAsync();
+        }
+    }
+
+    private async Task InvokeAgentAsync(AgentGroupChat chat, Agent agent, string input, string expected)
```
Comment (RogerBarreto) [inline dotnet/src/IntegrationTests/Agents/MixedAgentTests.cs:103]: Is not clear that Asserting is happening within this function, suggest adding `Assert` as a prefix.

```suggestion
    private async Task AssertInvokeAgentAsync(AgentGroupChat chat, Agent agent, string input, string expected)
```

File: dotnet/src/IntegrationTests/Agents/MixedAgentTests.cs [hunk 3]
```diff
@@ -0,0 +1,162 @@
...
+        IKernelBuilder kernelBuilder = Kernel.CreateBuilder();
+
+        kernelBuilder.AddOpenAIChatCompletion(
+            configuration.ModelId!,
+            configuration.ApiKey);
+
+        return kernelBuilder.Build();
+    }
+
+    public sealed class MenuPlugin
```
Comment (RogerBarreto) [inline dotnet/src/IntegrationTests/Agents/MixedAgentTests.cs:141]: If the scope of this plugin is just this test class, make it private.

```suggestion
    private sealed class MenuPlugin
```

**Review by RogerBarreto — APPROVED** [review]

File: dotnet/src/IntegrationTests/Agents/MixedAgentTests.cs [hunk 4]
```diff
@@ -0,0 +1,162 @@
...
+
+namespace SemanticKernel.IntegrationTests.Agents;
+
+#pragma warning disable xUnit1004 // Contains test methods used in manual verification. Disable warning for this file only.
+
+public sealed class MixedAgentTests
+{
+    private const string AssistantModel = "gpt-4o"; // Model must be able to support assistant API
+    private readonly IConfigurationRoot _configuration = new ConfigurationBuilder()
+            .AddJsonFile(path: "testsettings.json", optional: false, reloadOnChange: true)
```
[the remark this answers is not in the document]
Comment (crickman) [reply dotnet/src/IntegrationTests/Agents/MixedAgentTests.cs:23]: Sure.  This was copied from other integration tests and continues to exist in that form outside of this PR.

File: dotnet/src/IntegrationTests/Agents/MixedAgentTests.cs [hunk 5]
```diff
@@ -0,0 +1,162 @@
...
+    [Fact]
+    public async Task AzureOpenAIMixedAgentAsync()
+    {
+        AzureOpenAIConfiguration azureOpenAISettings = this._configuration.GetSection("AzureOpenAI").Get<AzureOpenAIConfiguration>()!;
+        Assert.NotNull(azureOpenAISettings);
+
+        await this.ExecuteAgentAsync(
+            this.CreateChatCompletionKernel(azureOpenAISettings),
+            new(azureOpenAISettings.ApiKey, azureOpenAISettings.Endpoint),
+            azureOpenAISettings.ChatDeploymentName!);
```
[the remark this answers is not in the document]
Comment (crickman) [reply dotnet/src/IntegrationTests/Agents/MixedAgentTests.cs]: This is calls into a centralized test method that includes action and asserting.  I've added a comment.

**Review by dmytrostruk — APPROVED** [review]

**Pull request merged**